overlord/devicestate/handlers_install.go

```diff
@@ -447,6 +493,23 @@ func (m *DeviceManager) doRestartSystemToRunMode(t *state.Task, _ *tomb.Tomb) er
 		return err
 	}
 
+	// will be used by the defer writeTimings() above
+	var chgIDs []string
+	var seedChg string
+	for _, chg := range st.Changes() {
+		if chg.Kind() == "seed" {
+			seedChg = chg.ID()
+		} else {
+			chgIDs = append(chgIDs, chg.ID())
+		}
+	}
+	st.Unlock()
+	err = writeTimings(boot.InstallHostWritableDir, chgIDs, seedChg)
```

Comment:
it's probably cleaner and safer to pass st, and move the loop on changes at the beginning, and then do a 

```
st.Unlock()
defer st.Lock()
```

inside writeTimings itself